perllib/FixMyStreet/App/Controller/Waste/Whitespace.pm

```diff
@@ -20,22 +20,86 @@ has log_ident => (
     }
 );
 
+has whitespace_config => (
+    is => 'rw',
+    isa => 'HashRef',
+    default => sub { {} }
+);
+
 sub receive_whitespace_event_notification : Path('/waste/whitespace') : Args(0) {
     my ($self, $c) = @_;
 
+    require SOAP::Lite;
+
+    $c->detach('/waste/echo/soap_error', [ 'Invalid method', 405 ]) unless $c->req->method eq 'POST';
+
+    $self->whitespace_config(FixMyStreet->config('COBRAND_FEATURES')->{whitespace}{bexley});
+
     my %headers = $c->req->headers->flatten;
     $self->log($c->req->method);
     $self->log(\%headers);
     $self->log($c->req->parameters);
     if ($c->req->body) {
         my $soap = join('', $c->req->body->getlines);
         $self->log($soap);
+        my $env = SOAP::Deserializer->deserialize($soap);
+        if ($env->valueof('//WorksheetPoke/secret') ne $self->whitespace_config->{missed_collection_secret}) {
+            return $c->detach('/waste/echo/soap_error', [ 'Unauthorized', 401 ]);
+        };
+        my %params;
+        $params{external_id} = $env->valueof('//WorksheetPoke/worksheetId');
+        $params{status} = $env->valueof('//WorksheetPoke/status');
+        $params{completed_dt} = $env->valueof('//WorksheetPoke/completedDate');
```

Comment:
This isn't a DateTime, it's just a string, I think the update code assumes in a couple of places it'll be a DateTime. `updated_datetime` is what the Echo code uses.